opt: split tpch stats quality tests

RaduBerinde opened the pull request:
opt: split tpch stats quality tests

Splitting the tpch stats quality tests into more manageable multiple
files, one per query. This allows re-running only particular queries.

Release note: None

----

opt: convert panics to errors for stats directive

Currently if a plan changes and a save table name is no longer
correct, we get a panic. The panic is annoying to work with (it stops
running other subtests). This change converts the panic in this case
to an error.

Release note: None

RaduBerinde committed:
opt: split tpch stats quality tests

Splitting the tpch stats quality tests into more manageable multiple
files, one per query. This allows re-running only particular queries.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42970)
<!-- Reviewable:end -->

RaduBerinde committed:
opt: convert panics to errors for stats directive

Currently if a plan changes and a save table name is no longer
correct, we get a panic. The panic is annoying to work with (it stops
running other subtests). This change converts the panic in this case
to an error.

Release note: None

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  Thanks for doing this!

> Reviewed 23 of 23 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42970#-:-LvLjQPRBWqu9BzxbEy2:bta3s4d)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
